ipa-core/src/protocol/ipa_prf/validation_protocol/validation.rs

```diff
@@ -469,4 +531,226 @@ pub mod test {
             assert_eq!(helper_1_left, helper_2_right);
         });
     }
+
+    /// This is a helper function to generate `u`, `v` values.
+    ///
+    /// outputs `(my_u_and_v, u_from_right_prover, v_from_left_prover)`
```

Comment:
So the paper says (on page 21):

"P_{i+1}, P_{i-1} compute vector u and v respectively as P_i does."

So I think this comment is wrong.

"u_from_right_prover" doesn't make sense. If the Prover (that is P_i) is to your right, then you are P_{i-1} in that context, meaning you should be computing *v*.